java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/docker/DefaultDockerfileGenerator.java

```diff
@@ -30,56 +29,55 @@
  */
 public class DefaultDockerfileGenerator implements DockerfileGenerator {
 
-  private static final String DOCKERFILE = "FROM %s\n"
-      + "ADD %s %s\n";
+  private static final String DOCKERFILE_FROM = "FROM %s\n";
+  private static final String DOCKERFILE_ADD_ARTIFACT = "ADD %s %s\n";
+  private static final String DOCKERFILE_JETTY_QUICKSTART = "RUN /scripts/jetty/quickstart.sh\n";
+
+  private static final String APP_DESTINATION = "$APP_DEST";
```

Comment:
I believe it does expand environment variables in ADD instructions - we use that currently by adding to the destination `$JETTY_BASE/webapps/root.war` for jetty. 

Each runtime can set the variable to rename the file as needed. From the [Dockerfile spec](https://docs.docker.com/engine/reference/builder/#add):
> If `dest` does not end with a trailing slash, it will be considered a regular file and the contents of `src` will be written at `dest`.